File: app/src/main/java/com/shveed/cookmegood/adapter/CartRecipeAdapter.java
package com.shveed.cookmegood.adapter;

import android.content.Context;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.shveed.cookmegood.entity.Recipe;
import com.shveed.wallpapperparser.R;

import java.util.List;

public class CartRecipeAdapter extends RecyclerView.Adapter<CartRecipeAdapter.ViewHolder> {

    LayoutInflater inflater;
    List<Recipe> recipes;

    public CartRecipeAdapter(Context context, List<Recipe> recipes) {
        this.recipes = recipes;
        this.inflater = LayoutInflater.from(context);
    }
    @Override
    public CartRecipeAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.item_cart_recipe, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(CartRecipeAdapter.ViewHolder holder, int position) {
        Recipe recipe = recipes.get(position);
        holder.imageView.setImageResource(R.drawable.pic1);
        holder.nameView.setText(recipe.getName());
    }

    @Override
    public int getItemCount() {
        return recipes.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        final ImageView imageView;
        final TextView nameView;
        ViewHolder(View view){
            super(view);
            imageView = (ImageView)view.findViewById(R.id.cartRecipeImage);
            nameView = (TextView) view.findViewById(R.id.cartRecipeName);
        }
    }
}

File: app/src/main/java/com/shveed/cookmegood/adapter/CartRecipeAdapter.kt
package com.shveed.cookmegood.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.shveed.cookmegood.data.dto.Recipe
import com.shveed.wallpapperparser.R
import kotlinx.android.synthetic.main.item_cart_recipe.view.*

class CartRecipeAdapter(private val recipes: List<Recipe>, val context: Context): RecyclerView.Adapter<CartRecipeAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
                LayoutInflater.from(context).inflate(
                        R.layout.item_cart_recipe,
                        parent, false
                )
        )
    }

    override fun getItemCount(): Int {
        return recipes.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        val recipe = recipes[position]

        holder.image.setImageResource(recipe.image!!.toInt())
        holder.name.text = recipe.name
    }

    class ViewHolder(view: View): RecyclerView.ViewHolder(view){
        val image = view.cartRecipeImage!!
        val name = view.cartRecipeName!!
    }
}